File: frontend/src/app/modules/models/customerModel.ts
import {WalletModel} from "./walletModel";

export class customerModel {
  id: number;
  name: string;
  surname: string;
  age: number;
  email: string;
  password: string;
  wallet: WalletModel;
  isActive: number;
  idLogInInf: number;

  static cloneBase(customer: customerModel): customerModel {
    const clonedCustomer: customerModel = new customerModel();
    clonedCustomer.name = customer.name;
    clonedCustomer.surname = customer.surname;
    clonedCustomer.age = customer.age;
    clonedCustomer.email = customer.email;
    clonedCustomer.password = customer.password;
    clonedCustomer.wallet = customer.wallet;
    clonedCustomer.isActive = customer.isActive;
    clonedCustomer.idLogInInf = customer.idLogInInf;
    return clonedCustomer;
  }
}

File: fapi/src/main/java/com/artsykov/fapi/models/CustomerModel.java
package com.artsykov.fapi.models;

import com.artsykov.fapi.entity.WalletEntity;
import lombok.Data;

import javax.validation.constraints.*;

@Data
public class CustomerModel {
    @Min(value=0, message="Incorrect customer id")
    private int id;

    @NotEmpty(message = "Name is required")
    @Pattern(regexp = "^[A-Z]{1}[a-z]+$",
            message = "Name is invalid")
    private String name;

    @NotEmpty(message = "Surname is required")
    @Pattern(regexp = "^[A-Z]{1}[a-z]+$",
            message = "Surname is invalid")
    private String surname;

    @Positive(message = "Age is invalid")
    private int age;

    @NotEmpty(message = "Email is required")
    @Email(message = "Email is incorrect")
    private String email;

    @Pattern(regexp="^[A-Za-z0-9]{8,}$",
            message="Password is invalid")
    private String password;
    private WalletModel wallet;

    @Min(value=0, message="Incorrect active status")
    @Max(value=1, message="Incorrect active status")
    private int isActive;

    @Min(value=0, message="Incorrect log in information id")
    private int idLogInInf;

    public CustomerModel() {}
}
